File: IRE/merge.py
import sys
out = open(sys.argv[3], 'w+')
with open(sys.argv[1]) as f1:
    with open(sys.argv[2]) as f2:
        line1 = f1.readline()
        line2 = f2.readline()
        word1 = line1.split(":")[0]
        word2 = line2.split(":")[0]
        while line1 != "" and line2 != "":
            if word1 < word2:
                out.write(line1)
                line1 = f1.readline()
                word1 = line1.split(":")[0]
            else:
                out.write(line2)
                line2 = f2.readline()
                word2 = line2.split(":")[0]
        while(line1 != ""):
            out.write(line1)
            word1 = line1.split(":")[0]
            line1 = f1.readline()

        while(line2 != ""):
            out.write(line2)
            word2 = line2.split(":")[0]
            line2 = f2.readline()

out.close()

File: IRE/merge.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    for (int i = 1; i <= 3; i += 2)
    {
        char filename1[100];
        char filename2[100];
        char outfilename[100];
        sprintf(filename1, "ind%d.txt", i);
        sprintf(filename2, "ind%d.txt", i + 1);
        sprintf(outfilename, "ind%d%d.txt", i, i + 1);
        ifstream infile1(filename1);
        ifstream infile2(filename2);
        ofstream outfile(outfilename);
        string line1, line2;
        string word1, word2;
        getline(infile1, line1);
        getline(infile2, line2);
        stringstream ss1(line1), ss2(line2);
        getline(ss1, word1, ':');
        getline(ss2, word2, ':');
        while (true)
        {
            cout << word1 << " " << word2 << endl;
            // stringstream ss1(line1), ss2(line2);
            if (word1.compare(word2))
            {
                outfile << line1 << endl;
                if (!getline(infile1, line1))
                {
                    break;
                }
                stringstream ss1(line1);
                getline(ss1, word1, ':');
            }
            else
            {
                outfile << line2 << endl;
                if (!getline(infile2, line2))
                {
                    break;
                }
                stringstream ss2(line2);
                getline(ss2, word1, ':');
            }
        }
    }
}